test/test_trixi.jl

```diff
@@ -98,3 +97,41 @@ macro test_nowarn_debug(expr)
     end
   end
 end
+
+# Modified version of `@test_nowarn` that prints the content of `stderr` when
+# it is not empty and ignnores module replacements.
+macro test_nowarn_mod(expr)
+  quote
+    let fname = tempname()
+      try
+        ret = open(fname, "w") do f
+          redirect_stderr(f) do
+            $(esc(expr))
+          end
+        end
+        stderr_content = read(fname, String)
+        if !isempty(stderr_content)
+          println("Content of `stderr`:\n", stderr_content)
+        end
+        test_successful = isempty(stderr_content)
+        # we also ignore simple module redefinitions for convenience
+        if !test_successful
+          only_module_replacements = true
+          for line in split(stderr_content, '\n')
+            if !isempty(line) && match(r"WARNING: replacing module .+\.", line) === nothing
+              only_module_replacements = false
+              break
+            end
+          end
+          if only_module_replacements
+            test_successful = true
+          end
+        end
+        @test test_successful
```

Comment:
```suggestion
        @test occursin(r"^(WARNING: replacing module .+\.\n)*$", read(fname, String))
```

Reply:
Wouldn't something like this be a lot easier and with a short comment also a lot more readable?

Reply:
Good suggestion :+1: